tools/clang_tidy.py

```diff
@@ -8,28 +8,35 @@
 import sys
 
 
-DEFAULT_FILE_PATTERN = r".*\.[ch](pp)?"
+DEFAULT_FILE_PATTERN = r".*\.c(c|pp)?"
 
 # @@ -start,count +start,count @@
 CHUNK_PATTERN = r"^@@\s+-\d+,\d+\s+\+(\d+)(?:,(\d+))?\s+@@"
 
 
-def run_shell_command(arguments, process_name=None):
+def run_shell_command(arguments, process_name=None, verbose=False):
     """Executes a shell command."""
     assert len(arguments) > 0
     try:
-        output = subprocess.check_output(arguments, stderr=subprocess.STDOUT)
-    except OSError:
-        _, e, _ = sys.exc_info()
+        if verbose:
+            output = subprocess.check_output(arguments, stderr=subprocess.STDOUT)
+        else:
+            with open(os.devnull, 'wb') as devnull:
+                output = subprocess.check_output(arguments, stderr=devnull)
+    except:  # noqa E722
         process_name = process_name or arguments[0]
-        raise RuntimeError("Error executing {}: {}".format(process_name, e))
+        _, error, _ = sys.exc_info()
+        message = "Error executing '{}'".format(process_name)
```

Comment:
It seems way more elegant to just print the command you're executing, before you execute it. Then you don't need to catch all.